conduit/pipeline.go

```diff
@@ -0,0 +1,299 @@
+package conduit
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"path/filepath"
+
+	"github.com/algorand/go-algorand-sdk/encoding/json"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/exporters"
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/plugins"
+	"github.com/algorand/indexer/processors"
+)
+
+func init() {
+	viper.SetConfigType("yaml")
+}
+
+const autoLoadParameterConfigName = "conduit.yml"
+
+type nameConfigPair struct {
+	Name   string                 `yaml:"Name"`
+	Config map[string]interface{} `yaml:"Config"`
+}
+
+// PipelineConfig stores configuration specific to the conduit pipeline
+type PipelineConfig struct {
+	ConduitConfig *Config
+
+	pipelineLogLevel string `yaml:"LogLevel"`
+	PipelineLogLevel log.Level
+	// Store a local copy to access parent variables
+	Importer   nameConfigPair   `yaml:"Importer"`
+	Processors []nameConfigPair `yaml:"Processors"`
+	Exporter   nameConfigPair   `yaml:"Exporter"`
+}
+
+// Valid validates pipeline config
+func (cfg *PipelineConfig) Valid() error {
```

Comment:
This can be private

Reply:
It might also be possible to set defaults and have some of it be automatic:
https://github.com/spf13/viper#establishing-defaults